benchmark/scripts/logging_metrics.py

```diff
@@ -0,0 +1,72 @@
+import keras
+
+
+class LoggingMetrics:
+    """Callback that save metrics to a logfile.
+
+    # Arguments
+        history_callback: instance of `keras.callbacks.History`.
+            Training parameters
+            (eg. batch size, number of epochs, loss, acc).
+        time_callback: instance of `keras.callbacks.Callback`.
+            Training parameters
+            (eg. time, time-step, speed).
+
+    # Raises
+        TypeError: In case of invalid object instance.
+    """
+
+    def __init__(self, history_callback, time_callback):
+        self.num_iteration = None
+        self.metric_list = []
+        self.pattern_list = []
+        self.retrieve_metrics(history_callback, time_callback)
+
+    def retrieve_metrics(self, history_callback, time_callback):
+        if not isinstance(history_callback, keras.callbacks.History):
+            raise TypeError('`history_callback` should be an instance of '
+                            '`keras.callbacks.History`')
+        if not isinstance(time_callback, keras.callbacks.Callback):
+            raise TypeError('`time_callback` should be an instance of '
+                            '`keras.callbacks.Callback`')
+
+        if hasattr(history_callback, 'epoch'):
+            self.metric_list.append(history_callback.epoch)
+            self.pattern_list.append('[Epoch %d]\t')
+
+        if hasattr(time_callback, 'times'):
+            self.metric_list.append(time_callback.get_time())
+            self.metric_list.append(time_callback.get_time_step())
+            self.metric_list.append(time_callback.get_speed())
+            self.pattern_list.append('time: %s\t')
+            self.pattern_list.append('time_step: %s\t')
+            self.pattern_list.append('speed: %s\t')
+
+        if 'loss' in history_callback.history:
```

Comment:
please move the metric constants to constant.py file & iterate over those to increase modularity